api/routers/service_accounts.py

```diff
@@ -1,121 +0,0 @@
-import logging
-import secrets
-import subprocess
-
-from fastapi import APIRouter, Response, status
-from pydantic import BaseModel
-
-router = APIRouter()
-logger = logging.getLogger("micro-auth")
-
-
-class KeyRequest(BaseModel):
-    username: str
-
-
-class ServiceAccountResponse(BaseModel):
-    access_key: str
-    secret_key: str
-
-
-class AccessKeyAndExpiry(BaseModel):
-    access_key: str
-    expiry: str
-
-
-class ServiceAccountListResponse(BaseModel):
-    service_accounts: list[AccessKeyAndExpiry]
-
-
-@router.post("/auth/minio/sa/")
-async def create_service_account(key_request: KeyRequest, response: Response) -> ServiceAccountResponse:
```

Comment:
how are these service accounts used?

Reply:
The service account is derived from a user and has a key/secret. The users can use the key and secret to access minio through the S3 protocol using the resource permissions shared with them in HydroShare.